zerver/models.py

```diff
@@ -1053,6 +1056,36 @@ def parse_usermessage_flags(val):
         mask <<= 1
     return flags
 
+class Attachment(models.Model):
+    MAX_FILENAME_LENGTH = 100
+    file_name = models.CharField(max_length=MAX_FILENAME_LENGTH, db_index=True)
+    path_id = models.TextField(db_index=True)
+    owner = models.ForeignKey(UserProfile)
+    message = models.ForeignKey(Message, blank=True, null=True)
+    create_time = models.DateTimeField(default=timezone.now, db_index=True)
+
+    def __repr__(self):
+        return (u"<Attachment: %s>" % (self.file_name))
+
+    def is_claimed(self):
+        return self.message is not None
+
+    def get_url(self):
+        return "/user_uploads/%s" % (path_id)
+
+def get_attachments_by_owner_id(uid):
+    return Attachment.objects.filter(owner=uid).select_related('owner')
+
+def get_owners_from_file_name(file_name):
+    #The returned vaule will list of owners since different users can upload
+    #same file
+    return Attachment.objects.filter(file_name=file_name).select_related('owner')
+
+def get_old_unclaimed_attachments():
+    one_week_ago = timezone.now() - datetime.timedelta(weeks=1)
+    old_attachments = Attachment.objects.filter(message=None, create_time__lt=one_week_ago)
+    return old_attachments
```

Comment:
Let's include a new management command that calls this function, under zerver/management/commands

Also, can we add a test that this function behaves correctly?  I'm imagining creating a couple Attachment objects, setting their age to 2 weeks ago, one with a message and the other not, and then confirming that just the one without a message gets deleted when we call this.